From b44c3a61ca756a248656fc2f14ae692510aa2743 Mon Sep 17 00:00:00 2001
From: De Huo <De.Huo@windriver.com>
Date: Tue, 8 Oct 2019 09:57:58 +0800
Subject: [PATCH] Rule file_permissions_sshd_private_key update remediate script

If there is no keys in /etc/ssh generate the keys in the remediate script.

Upstream-Status: Pending

Signed-off-by: De Huo <De.Huo@windriver.com>
---
 .../bash/wrlinux1019.sh                            | 22 ++++++++++++++++++++++
 1 file changed, 22 insertions(+)

diff --git a/linux_os/guide/services/ssh/file_permissions_sshd_private_key/bash/wrlinux1019.sh b/linux_os/guide/services/ssh/file_permissions_sshd_private_key/bash/wrlinux1019.sh
index 78395dd..20f237e 100644
--- a/linux_os/guide/services/ssh/file_permissions_sshd_private_key/bash/wrlinux1019.sh
+++ b/linux_os/guide/services/ssh/file_permissions_sshd_private_key/bash/wrlinux1019.sh
@@ -1,3 +1,25 @@
 # platform = multi_platform_wrlinux
 
+SYSCONFDIR=/etc/ssh
+
+if [ ! -d $SYSCONFDIR ]; then
+	/bin/mkdir -p $SYSCONFDIR
+fi
+
+if [ ! -f /etc/ssh/ssh_host_rsa_key ]; then
+	/usr/bin/ssh-keygen -q -f ${SYSCONFDIR}/ssh_host_rsa_key -N '' -t rsa 
+fi
+
+if [ ! -f /etc/ssh/ssh_host_dsa_key ]; then
+	/usr/bin/ssh-keygen -q -f ${SYSCONFDIR}/ssh_host_dsa_key -N '' -t dsa 
+fi
+
+if [ ! -f /etc/ssh/ssh_host_ecdsa_key ]; then
+	/usr/bin/ssh-keygen -q -f ${SYSCONFDIR}/ssh_host_ecdsa_key -N '' -t ecdsa
+fi
+
+if [ ! -f /etc/ssh/ssh_host_ed25519_key ]; then
+	/usr/bin/ssh-keygen -q -f ${SYSCONFDIR}/ssh_host_ed25519_key -N '' -t ed25519
+fi
+
 chmod 0600 /etc/ssh/*_key
-- 
1.9.1

